xlsxwriter/xmlwriter.py

```diff
@@ -71,12 +69,17 @@ def _xml_end_tag(self, tag):
         # Write an XML end tag.
         self.fh.write("</%s>" % tag)
 
-    def _xml_empty_tag(self, tag, attributes=[]):
-        # Write an empty XML tag with optional attributes.
+    def _build_escaped_attribute_list(self, attributes):
```

Comment:
Is _build_escaped_attribute_list() a performance or aesthetic change? If it is the latter then omit it.

Reply:
it's separated so that experimentation can occur.  In some versions of python different ways of concatenating strings are more performant.  I tried a few different kinds before settling on this one.  I'm not sure if we are heading down that road, but it certainly was a lot of boilerplate code being merged into a single method.  Let me know and I can omit these changes.

Reply:
Omit this change. The boilerplate is to avoid a function call. But I don't know if that is much slower. Either way if the change doesn't improve performance leave it out (of this patchset which should be restricted to performance).

Reply:
yes, a function call will be slower.  However, if we leave this change in, you will be able to experiment with performance.  I think this block could be easily replaced with some c code or even use an existing native python library (I'm not sure which one, but urlparse comes to mind) or some other xml library if the user has it installed to speed things up.  This refactor I think is worth the cost of a single function call, if only to make the code more readable.